src/Core/Checkout/Document/Service/DocumentGenerator.php

```
@@ -325,24 +328,30 @@ private function ensureDocumentMediaFileGenerated(DocumentEntity $document, stri
         return $document;
     }
 
-    private function resolveMediaId(DocumentGenerateOperation $operation, Context $context, RenderedDocument $document): ?string
+    private function resolveMediaId(DocumentGenerateOperation $operation, Context $context, RenderedDocument $document, ?string $documentType = null, ?RendererResult $result = null): ?string
     {
         if ($operation->isStatic()) {
             return null;
         }
 
-        try {
-            $blob = $this->fileRendererRegistry->render($document);
-        } catch (\Throwable) {
-            return null;
+        if (!Feature::isActive('v6.7.0.0')) {
+            try {
+                $document->setContent($this->fileRendererRegistry->render($document));
+
+                if ($documentType && $result) {
+                    $this->rendererRegistry->finalize($documentType, [$operation->getOrderId() => $operation], $context, new DocumentRendererConfig(), $result);
+                }
```

Is there a reason why we should silence any exceptions from the `finalize` step?

---

Changed the `finalize` a little bit
https://github.com/shopware/shopware/pull/6470/commits/f91303828b5eff9d723762611c7de2cd536c8811